Upgrade CI images to Node 16

Blocks #1695

WIP branch to see what breaks: https://github.com/votingworks/vxsuite/tree/tara/node-16-upgrade-ci-images

- [x] Upgrade CI images to 16.14.2 52735b2e78c7ec4d947887d3a0b295761e6cb271
- [x] Fix failing `node-canvas` installs (upgraded to ^2.9.0 https://github.com/Automattic/node-canvas/issues/1511) e7e8a61c0aefba1437fefcd32e0a7da7a27f820b
- [x] Fix failing `@votingworks/qrdetect` install, the issue seems to be with `zbar`
  - `../addon/zbar/processor/x.c:28:10: fatal error: X11/Xlib.h: No such file or directory`
 - [x] Fix missing `libpng`
 - [x] Fix missing `libjpeg`
 - [x] #1716
   - I can reproduce this locally, and I've confirmed that `ts-jest` is installed with `pnpm --dir libs/res-to-ts ls ts-jest`. I've also tried uninstalling and re-installing `jest`, `ts-jest`, and `@types/jest`.
  - [x] #1718
  - [x] Fix failure when downloading Cypress binary in `test-libs-fixtures`
    ```
    URL: https://download.cypress.io/desktop/4.12.1?platform=linux&arch=x64
    .../cypress@4.12.1/node_modules/cypress postinstall: Error: Failed downloading the Cypress binary.
    .../cypress@4.12.1/node_modules/cypress postinstall: Response code: 52
    ```
- [x] Fix failing scan unit tests https://github.com/votingworks/vxsuite/issues/1696
- [ ] #1726
- [x] #1727
- [x] #1729

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -589,7 +589,10 @@ commands:
             "pnpm-lock.yaml" }}
       - run:
           name: Setup Dependencies
-          command: pnpm install --frozen-lockfile
+          command: |
+            sudo apt update -y
+            sudo apt install libx11-dev
+            pnpm install --frozen-lockfile
       - save_cache:
           key:
             dotcache-cache-{{checksum ".circleci/config.yml" }}-{{ checksum

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,9 +1,12 @@
 version: 2.1
 
+orbs:
+  browser-tools: circleci/browser-tools@1.1.0
+
 executors:
   nodejs-browsers:
     docker:
-      - image: cimg/node:16.14.2-browsers
+      - image: cimg/python:3.9.12-browsers
   nodejs:
     docker:
       - image: cimg/node:16.14.2
@@ -62,28 +65,6 @@ jobs:
           name: Test
           command: |
             pnpm --dir frontends/bmd test
-      - store_artifacts:
-          path: frontends/bmd/cypress/screenshots
-      - store_artifacts:
-          path: frontends/bmd/cypress/videos
-
-  test-integration-testing-bmd:
-    executor: nodejs-browsers
-    resource_class: xlarge
-    steps:
-      - checkout-and-install
-      - run:
-          name: Build
-          command: |
-            pnpm --dir frontends/bmd build
-      - run:
-          name: Lint
-          command: |
-            pnpm --dir frontends/bmd lint
-      - run:
-          name: Test
-          command: |
-            pnpm --dir frontends/bmd test:e2e
       - store_artifacts:
           path: frontends/bmd/cypress/screenshots
       - store_artifacts:
@@ -151,11 +132,34 @@ jobs:
           command: |
             pnpm --dir frontends/precinct-scanner test
 
-  test-integration-testing-bsd:
-    docker:
-      - image: circleci/python:3.9.2-node-browsers
+  test-integration-testing-bmd:
+    executor: nodejs-browsers
     resource_class: xlarge
     steps:
+      - install-cypress-browser
+      - checkout-and-install
+      - run:
+          name: Build
+          command: |
+            pnpm --dir frontends/bmd build
+      - run:
+          name: Lint
+          command: |
+            pnpm --dir frontends/bmd lint
+      - run:
+          name: Test
+          command: |
+            pnpm --dir frontends/bmd test:e2e
+      - store_artifacts:
+          path: frontends/bmd/cypress/screenshots
+      - store_artifacts:
+          path: frontends/bmd/cypress/videos
+
+  test-integration-testing-bsd:
+    executor: nodejs-browsers
+    resource_class: xlarge
+    steps:
+      - install-cypress-browser
       - checkout-and-install
       - run:
           name: Build
@@ -175,10 +179,10 @@ jobs:
           path: integration-testing/bsd/cypress/videos
 
   test-integration-testing-election-manager:
-    docker:
-      - image: circleci/python:3.9.2-node-browsers
+    executor: nodejs-browsers
     resource_class: xlarge
     steps:
+      - install-cypress-browser
       - checkout-and-install
       - run:
           name: Build
@@ -599,3 +603,8 @@ commands:
             "pnpm-lock.yaml" }}
           paths:
             - ~/.pnpm-store
+  install-cypress-browser:
+    description: Installs a browser for Cypress tests.
+    steps:
+      - browser-tools/install-chrome
+      - browser-tools/install-chromedriver

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -3,10 +3,10 @@ version: 2.1
 executors:
   nodejs-browsers:
     docker:
-      - image: circleci/node:12.19.0-browsers
+      - image: cimg/node:16.14.2-browsers
   nodejs:
     docker:
-      - image: circleci/node:12.19
+      - image: cimg/node:16.14.2
 
 jobs:
   test-codemods:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -591,7 +591,7 @@ commands:
           name: Setup Dependencies
           command: |
             sudo apt update -y
-            sudo apt install libx11-dev
+            sudo apt install libx11-dev libpng-dev
             pnpm install --frozen-lockfile
       - save_cache:
           key:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -591,7 +591,7 @@ commands:
           name: Setup Dependencies
           command: |
             sudo apt update -y
-            sudo apt install libx11-dev libpng-dev
+            sudo apt install libx11-dev libpng-dev libjpeg-dev
             pnpm install --frozen-lockfile
       - save_cache:
           key:

--- a/services/converter-ms-sems/Makefile
+++ b/services/converter-ms-sems/Makefile
@@ -5,7 +5,7 @@ install-python:
 install: install-python
 
 install-dependencies:
-	sudo python3 -m pip install pipenv
+	python3 -m pip install pipenv || sudo python3 -m pip install pipenv
 	python3 -m pipenv install
 
 install-dev-dependencies:

--- a/services/scan/src/cli/mock-scanner/index.test.ts
+++ b/services/scan/src/cli/mock-scanner/index.test.ts
@@ -6,6 +6,10 @@ beforeEach(() => {
   nock.cleanAll();
 });
 
+afterAll(() => {
+  nock.enableNetConnect();
+});
+
 test('no command error', async () => {
   jest.spyOn(process.stderr, 'write').mockReturnValue(true);
   expect(await mockScanner([])).toEqual(-1);

